### PR TITLE
add documentation for for loop

### PR DESCRIPTION
**Pull request by aalok-t**
### Description

Update move book to include for loops

### Checklist
- [x] Do all Lints pass?
- [x] Have you ran `pnpm spellcheck`?
- [x] Have you ran `pnpm fmt`?
- [x] Have you ran `pnpm lint`?

### CODE REVIEW

File: docs/move/book/loops.md [hunk 1]
```diff
@@ -114,6 +114,45 @@ fun pick(
...
+fun sum(n: u64): u64 {
+    let sum = 0;
+    for (i in 0..n) {
+        sum = sum + i;
+    };
+
+    sum
+}
+```
+
```
Comment (brmataptos) [inline docs/move/book/loops.md:133]: Suggest adding some details:

The loop index variable (`i` in this example) currently must be a numeric type (inferred from the bounds), and the bounds `0` and `n` here can be replaced by arbitrary numeric expressions.  Each is only evaluated once, at the start of the loop.  The index variable `i` is assigned the starting bound `0` and incremented after each loop iteration; when the index `i` reaches or exceeds the upper bound than the loop exits.

File: docs/move/book/loops.md [hunk 2]
```diff
@@ -114,6 +114,45 @@ fun pick(
...
+### `break` and `continue` in `for` loops
+
+Similar to `while` loops, the `break` expression can be used in `for` loops to exit prematurely. The `continue` expression can be used to skip the current iteration and move to the next. Here's an example that demonstrates the use of both `break` and `continue`. The loop will iterate through numbers from `0` to `n`, summing up them up. It will skip numbers that are divisible by `3` (using `continue`) and stop when it encounters a number greater than `10` (using `break`):
+
+```move
+fun sum_conditional(n: u64): u64 {
+    let sum = 0;
+    for (i in 0..(n + 1)) {
```
Comment (brmataptos) [inline docs/move/book/loops.md:141]: Use a different variable than `i` here, to show noobs that the index variable can have any name.  Perhaps `count` or `idx`.

**Review by brmataptos — COMMENTED** [review]
No reason not to be a little more complete here.

File: docs/move/book/loops.md (hunk 1 above)
Comment (aalok-t) [reply docs/move/book/loops.md:133]: Added the description.  I am calling it an iterator as it is not necessarily a loop index (for example when lower bound is non-zero).

File: docs/move/book/loops.md (hunk 2 above)
Comment (aalok-t) [reply docs/move/book/loops.md]: I used `iter`

File: docs/move/book/loops.md [hunk 3]
```diff
@@ -160,12 +201,13 @@ fun sum_intermediate(n: u64): u64 {
 }
 ```
 
-## The type of `while` and `loop`
+## The type of `while`, `loop`, and `for` expression
 
-Move loops are typed expressions. A `while` expression always has type `()`.
+Move loops are typed expressions. The `while` and `for` expression always has type `()`.
```
Comment (brmataptos) [inline docs/move/book/loops.md:206]:
```
The `while` and `for` expressions always have type `()`.
                                ^ ^^^^^^ ^^^^
```

**Review by brmataptos — APPROVED** [review]

**Review by vineethk — APPROVED** [review]

**Pull request merged**